core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -180,6 +211,54 @@ public String getDisplayName() {
         }
     }
 
+    private void createInitialApiToken(User user) throws IOException, InterruptedException {
+        ApiTokenProperty apiTokenProperty = user.getProperty(ApiTokenProperty.class);
+
+        String sysProp = ADMIN_INITIAL_API_TOKEN;
+        if (sysProp.equals("true")) {
+            TokenUuidAndPlainValue tokenUuidAndPlainValue = apiTokenProperty.generateNewToken("random-generation-during-setup-wizard");
+            FilePath fp = getInitialAdminApiTokenFile();
+            // same comment as in the init method
+
+            // JENKINS-33599 - write to a file in the jenkins home directory
+            // most native packages of Jenkins creates a machine user account 'jenkins' to run Jenkins,
+            // and use group 'jenkins' for admins. So we allow groups to read this file
+            fp.touch(System.currentTimeMillis());
+            fp.chmod(0640);
```

Comment:
insecure on windows?

Reply:
As secure as https://github.com/jenkinsci/jenkins/blob/5f8b0d838fade631220f45a3f86ce67e930edcec/core/src/main/java/jenkins/install/SetupWizard.java#L161 ;)

(FTR the chmod method is ignored on Windows)

Reply:
that was my point this does nothing on windows and things have moved on in the Java world since this was cerated and you can create secure files on windows.

You could even do this in the old days with `File.createTempFile()` which ensured the file was secure by default....

yes there may be oher places that are not good - but as this is new code we should do better from a security PoV should we not?

Reply:
I doubt we really care. How many people these days have a multi-user OS running a Jenkins service under one account which is not intended to be accessible to other accounts?

Anyway, check whether it suffices to use `File` (no need for `FilePath` for an on-controller file) and something like

```
f.setReadable(true, false);
```

which would be easier than the `java.nio.file` versions.

Reply:
As there is no difference with existing code and having two different approaches will lead to more complicated time when (and if) we will want to correct it, I will let the code like this for the moment.

If you have any suggestion, a follow-up PR is welcome.